test/E2ETest/TyePurgeTests.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Tye;
+using Microsoft.Tye.ConfigModel;
+using Microsoft.Tye.Hosting;
+using Microsoft.Tye.Hosting.Model;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace E2ETest
+{
+    public class TyePurgeTests
+    {
+        private static readonly int MaxRetries = 5;
+
+        private readonly ITestOutputHelper output;
+        private readonly TestOutputLogEventSink sink;
+
+        public TyePurgeTests(ITestOutputHelper output)
+        {
+            this.output = output;
+            sink = new TestOutputLogEventSink(output);
+        }
+
+        [Fact]
+        public async Task FrontendBackendPurgeTest()
+        {
+            var projectDirectory = new DirectoryInfo(Path.Combine(TestHelpers.GetSolutionRootDirectory("tye"), "samples", "frontend-backend"));
+            var tempDirectory = TempDirectory.Create();
+            DirectoryCopy.Copy(projectDirectory.FullName, tempDirectory.DirectoryPath);
+
+            var projectFile = new FileInfo(Path.Combine(tempDirectory.DirectoryPath, "tye.yaml"));
+            var tyeDir = new DirectoryInfo(Path.Combine(tempDirectory.DirectoryPath, ".tye"));
+            using var host = new TyeHost(ConfigFactory.FromFile(projectFile).ToHostingApplication(), Array.Empty<string>())
```

Comment:
There will be some slight conflicts here due to my changes, but it should be copyable from other tests.